python_modules/dagster/dagster/core/execution.py

```diff
@@ -331,27 +343,32 @@ def _gen_thing():
 
 
 @contextmanager
-def as_ensured_single_gen(thing_or_gen):
+def user_code_context_manager(user_fn, error_cls):
```

Comment:
i think we should pass up an error context str to pass to the error boundary from the caller of this

Reply:
we should also maybe indicate in the name that this *might* be a context manager. i was never really happy with the nomenclature i came up with